src/docker/ContextManager.ts

```diff
@@ -180,34 +180,40 @@ export class DockerContextManager implements ContextManager, Disposable {
 
             ext.treeInitError = undefined;
 
-            let dockerHost: string | undefined;
+            let dockerHostEnv: string | undefined;
+            let dockerContextEnv: string | undefined;
             const config = workspace.getConfiguration('docker');
-            if ((dockerHost = config.get('host'))) { // Assignment + check is intentional
+            if ((dockerHostEnv = config.get('host'))) { // Assignment + check is intentional
                 actionContext.telemetry.properties.hostSource = 'docker.host';
-            } else if ((dockerHost = process.env.DOCKER_HOST)) { // Assignment + check is intentional
+            } else if ((dockerHostEnv = process.env.DOCKER_HOST)) { // Assignment + check is intentional
                 actionContext.telemetry.properties.hostSource = 'env';
+            } else if ((dockerContextEnv = config.get('context'))) { // Assignment + check is intentional
+                actionContext.telemetry.properties.hostSource = 'docker.context';
+            } else if ((dockerContextEnv = process.env.DOCKER_CONTEXT)) { // Assignment + check is intentional
+                actionContext.telemetry.properties.hostSource = 'envContext';
             } else if (!(await fse.pathExists(dockerContextsFolder)) || (await fse.readdir(dockerContextsFolder)).length === 0) {
                 // If there's nothing inside ~/.docker/contexts/meta, then there's only the default, unmodifiable DOCKER_HOST-based context
                 // It is unnecessary to call `docker context inspect`
                 actionContext.telemetry.properties.hostSource = 'defaultContextOnly';
-                dockerHost = isWindows() ? WindowsLocalPipe : UnixLocalPipe;
+                dockerHostEnv = isWindows() ? WindowsLocalPipe : UnixLocalPipe;
             } else {
-                dockerHost = undefined;
+                dockerHostEnv = undefined;
             }
 
-            if (dockerHost !== undefined) {
-                actionContext.telemetry.properties.hostProtocol = new URL(dockerHost).protocol;
+            if (dockerHostEnv !== undefined) {
+                actionContext.telemetry.properties.hostProtocol = new URL(dockerHostEnv).protocol;
+                await this.setVsCodeContext('vscode-docker:contextLocked', true);
 
                 return [{
                     ...defaultContext,
                     Current: true,
-                    DockerEndpoint: dockerHost,
+                    DockerEndpoint: dockerHostEnv,
                 } as DockerContext];
             }
 
             // No value for DOCKER_HOST, and multiple contexts exist, so check them
             const result: DockerContext[] = [];
-            const { stdout } = await execAsync('docker context ls --format="{{json .}}"', ContextCmdExecOptions);
+            const { stdout } = await execAsync('docker context ls --format="{{json .}}"', { ...ContextCmdExecOptions, env: { ...process.env, DOCKER_CONTEXT: dockerContextEnv } });

```

Comment:
This code relies on the fact that `docker context ls --context foo` will report "foo" as current context regardless what the ambient context is. This is exactly what we want, but it is not super obvious. Consider adding a comment.

Reply:
Good idea. Originally I had the logic where it would list them and if `dockerContextEnv` was set, compare names--but I realized that was redundant because the CLI already has the right behavior from the get go. A comment is definitely a good idea here.